src/Atmos/Model/reconstructions.jl

```diff
@@ -0,0 +1,82 @@
+export AtmosFVConstantBalanced
+
+using ..DGMethods.FVReconstructions: AbstractReconstruction, FVConstant
+using StaticArrays: SUnitRange
+import ..DGMethods.FVReconstructions: width
+
+struct AtmosFVConstantBalanced{M} <: AbstractReconstruction
+    atmos::M
+end
+width(::AtmosFVConstantBalanced) = 1
+
+function (recon::AtmosFVConstantBalanced)(
+    state_bot,
+    state_top,
+    cell_states::SVector{3},
+    cell_aux::SVector{3},
+    cell_weights,
+)
+    FT = eltype(state_bot)
+    cell_state = cell_states[2]
+
+    FVConstant()(
+        state_bot,
+        state_top,
+        cell_states[SUnitRange(2, 2)],
+        cell_aux,
+        cell_weights,
+    )
+
+    m = recon.atmos
+    hydrostatic_reconstruction!(
+        Vars{vars_state(m, Primitive(), FT)}(state_bot),
+        Vars{vars_state(m, Primitive(), FT)}(state_top),
+        Vars{vars_state(m, Prognostic(), FT)}(cell_state),
+        Vars{vars_state(m, Auxiliary(), FT)}.(cell_aux),
+    )
+end
```

Comment:
I think we can use g, for the hydrostatic balance reconstruction, then we do not need extra aux, especially when we are at the boundary.

Do we compute Φ as g * altitude ? I think we can compute Φ as g * altitude and from my 2D experiments, I use this and only observe computing \nabla Φ by DG derivative instead g r makes it more accurate